contracts/Tokens/VAI/VAIController.sol

```diff
@@ -0,0 +1,759 @@
+
+        emit NewVAIReceiver(receiver, newReceiver);
+        receiver = newReceiver;
+    }
+
+    /**
+     * @notice Set VAI mint cap
+     * @param _mintCap the amount of VAI that can be minted
+     * @custom:event NewVAIMintCap emits on success
+     * @custom:access Controlled by AccessControlManager
+     */
+    function setMintCap(uint256 _mintCap) external {
+        _checkAccessAllowed("setMintCap(uint256)");
+
+        emit NewVAIMintCap(mintCap, _mintCap);
+        mintCap = _mintCap;
+    }
+
+    /**
+     * @notice Set Treasury Details
+     * @param newTreasuryGuardian new TreasuryGuardian address
+     * @param newTreasuryAddress new Treasury address
+     * @param newTreasuryPercent new Treasury percentage
+     * @custom:error ZeroAddressNotAllowed is thrown if newTreasuryGuardian is zero address
+     * @custom:error ZeroAddressNotAllowed is thrown if newTreasuryAddress is zero address
+     * @custom:error CallerNotAuthorized is thrown if caller is not owner or treasury guardian
+     * @custom:error TreasuryPercentOverflow is thrown if newTreasuryPercent exceeds 1e18
+     * @custom:event NewTreasuryGuardian emits on success
+     * @custom:event NewTreasuryAddress emits on success
+     * @custom:event NewTreasuryPercent emits on success
+     * @custom:access Controlled by Governance and TreasuryGuardian
+     */
+    function setTreasuryData(
+        address newTreasuryGuardian,
+        address newTreasuryAddress,
+        uint256 newTreasuryPercent
+    ) external {
+        // Check caller is owner
+        if (!(msg.sender == owner() || msg.sender == treasuryGuardian)) {
+            revert CallerNotAuthorized();
+        }
+
+        ensureNonzeroAddress(newTreasuryGuardian);
+        ensureNonzeroAddress(newTreasuryAddress);
+
+        if (newTreasuryPercent > 1e18) {
+            revert TreasuryPercentOverflow();
+        }
+
+        emit NewTreasuryGuardian(treasuryGuardian, newTreasuryGuardian);
+        emit NewTreasuryAddress(treasuryAddress, newTreasuryAddress);
+        emit NewTreasuryPercent(treasuryPercent, newTreasuryPercent);
+
+        treasuryGuardian = newTreasuryGuardian;
+        treasuryAddress = newTreasuryAddress;
+        treasuryPercent = newTreasuryPercent;
+    }
+
+    /**
+     * @notice Accrues Vai Interest
+     */
+    function accrueVAIInterest() public {
+        uint256 delta = getVAIRepayRatePerBlock() * (_getBlockNumber() - accrualBlockNumber);
+        delta += VAIMintIndex;
+
+        VAIMintIndex = delta;
+        accrualBlockNumber = _getBlockNumber();
+    }
+
+    /**
+     * @notice This function calculates the VAI amount that can be minted to the minter
+     * @param minter Address of the minter
+     * @custom:error ZeroValueNotAllowed is thrown if oraclePrice is 0
+     * @return (uint256) Returns the mintable VAI amount
+     */
+    // solhint-disable-next-line code-complexity
+    function getMintableVAI(address minter) public view returns (uint256) {
```

Comment:
Is there a case when mintable VAI != VAI mint rate * borrowing power? If not, maybe we could just do sth like

```
(, uint256 borrowingPower, ,) = comptroller.getBorrowingPower(minter);
return VAIMintRate * borrowingPower;
```